api/pkg/apis/v1alpha1/vendors/solution-vendor.go

```diff
@@ -245,14 +268,29 @@ func (c *SolutionVendor) onReconcile(request v1alpha2.COARequest) v1alpha2.COARe
 				Body:  []byte(err.Error()),
 			})
 		}
-		delete := request.Parameters["delete"]
+		isRemove := request.Parameters["delete"]
 		targetName := ""
 		if request.Metadata != nil {
 			if v, ok := request.Metadata["active-target"]; ok {
 				targetName = v
 			}
 		}
-		summary, err := c.SolutionManager.Reconcile(ctx, deployment, delete == "true", namespace, targetName)
+
+		instance := deployment.Instance.ObjectMeta.Name
+		sLog.InfofCtx(ctx, "V (Solution): onReconcile create context with timeout, instance: %s, job id: %s, isRemove: %s", instance, deployment.JobID, isRemove)
+		cancelCtx, cancel := context.WithTimeout(ctx, defaultTimeout)
+		if isRemove != "true" {
+			c.SolutionManager.AddCancelFunc(ctx, namespace, instance, deployment.JobID, cancel)
+		}
+		defer func() {
+			log.InfofCtx(rContext, "V (Solution): onReconcile complete, namespace: %s, instance: %s, job id: %s, isRemove: %s", namespace, instance, deployment.JobID, isRemove)
+			cancel()
+			if isRemove != "true" {
+				c.SolutionManager.UntrackJob(rContext, namespace, instance, deployment.JobID)
+			}
```

Comment:
I would prefer moving logics regarding JobId tracking and cancelFunc to solution manager. The logic in the vendor should be as simple as possible